[ie/tvplay]: fix broken extractor

lukassup opened the pull request:
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

<!--

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible

-->

Fixed TVPlay extractor as it was broken likely due to site being updated.  


<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--

# PLEASE FOLLOW THE GUIDE BELOW

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
- Use *Preview* tab to see how your *pull request* will actually look like

-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

bashonly commented on yt_dlp/extractor/tvplay.py:
```diff
@@ -216,8 +216,8 @@ class TVPlayHomeIE(InfoExtractor):
             play\.(?:tv3|skaties)\.(?P<country>lv|lt|ee)/
-            (?P<live>lives/)?
-            [^?#&]+(?:episode|programme|clip)-(?P<id>\d+)
+            (lives/)?
+            [^?#&]+(?P<category>episode|programme|clip|live)-(?P<id>\d+)
```
```suggestion
            [^?#&]+,(?P<category>episode|programme|clip|live)-(?P<id>\d+)
```

bashonly commented on yt_dlp/extractor/tvplay.py:
```diff
@@ -263,18 +263,30 @@ class TVPlayHomeIE(InfoExtractor):
 
     def _real_extract(self, url):
-        country, is_live, video_id = self._match_valid_url(url).groups()
+        country, category, video_id = self._match_valid_url(url).groups()
```
```suggestion
        country, category, video_id = self._match_valid_url(url).group('country', 'category', 'id')
```

lukassup commented:
@bashonly thanks for the fixes - tested those locally (in my own script calling yt_dlp api) and they work!

btw, can you share the command to run test cases in `_TESTS`? `hatch test` produces `no tests ran in 0.43s`

bashonly commented:
You would run `hatch test TVPlayHome` or `hatch test TVPlayHome_1` or `hatch test TVPlayHome_2` or `hatch test TVPlayHome_all` etc, but there's a problem: every test case in this extractor is either dead (the 1st), marked `skip` (the 2nd), or `only_matching` (the rest of these -- they will not be counted towards the extractor tests, only the YoutubeDL matching test). Could you add some working tests?

lukassup commented:
> Could you add some working tests?
Not sure to be honest. Non-live stream is DRM protected now so I'm not sure how to handle that with yt-dlp

lukassup commented:
> Could you add some working tests?

Checking but not sure so far to be honest. Looks like most of non-live content is DRM-protected and live streams are likely to break eventually. We could probably leave them as is?

lukassup commented:
Added some tests that are working for now - I could not find a way to identify whether the stream has DRM or not so I added `'skip_download': 'm3u8'` to those tests.

lukassup commented:
@bashonly let me know if there is anything else I'd need to address

gamer191 commented:
> Added some tests that are working for now - I could not find a way to identify whether the stream has DRM or not so I added `'skip_download': 'm3u8'` to those tests.

You can find out if a stream is drm-protected by checking whether yt-dlp downloads it. If you wanna be extra safe, you should also check if the resulting video is playable

If you can not find any drm-free streams, the extractor will most likely have to be removed (ultimately not my decision). Similarly, if you can not find any DRM-free VODs then imo the extractor should be limited to live streams (the `else` statement should be removed from any `if is_live else` statements)
